### ISSUE TITLE
Problems with mustache values/helpers in widgets

### ISSUE THREAD

**Issue by karfau** [ruhoh/ruhoh.rb#112]
I just tried to use the `?to_json`-Helper inside a widget-layout:

```
    {{ config?to_json }}
```

This always returns an "empty text".
using any property from the config.yml directly works as expected.

Are the helpers / is this helper available for the widgets?

The other thing I recognized `config` seems to be the only payload given to the widget.
I would like to include  

```
    {{ urls.media }}
```

so that a path in the config.yml can be relative.

Any ideas about all this?

**Comment by plusjade** [comment]
@karfau you are right, the entire environment is _not_ passed to the widget context. It was done that way for simplicity's-sake but does present problems. 

I plan on overhauling the widget system in ruhoh 2.0. which I now finally working on documentation for. Hope to have it out over holiday break.

**Comment by karfau** [comment]
When you are talking about "having the documentation out over the holiday break", and looking at the existing branches, it looks like you are talking about documentation on existing 2.0 code. _cool_

I really like this project and if I can be of any help for contribution (be it 1.1 or 2.0), just ask.
(But keep in mind, that my ruby-"experience" starts with looking at the code you wrote ;) )

I just entered the maillinglist to stay up-to-date about the project.

How far is the 2.0 alpha branch compared to existing features compared to 1.1?

**Comment by plusjade** [comment]
> How far is the 2.0 alpha branch compared to existing features compared to 1.1?

It is a complete rewrite of the codebase. So internally it is completely different (I'd like to say better). Refreshingly though, the core public API is pretty similar.  The mustache methods have changed _a little_. They are indeed different but I think they are more logical, cleaner and make way for much easier level of customizability.

I'll save trying to explain everything and just hope the new way makes more intuitive sense.

> I really like this project and if I can be of any help for contribution (be it 1.1 or 2.0), just ask.

The best thing you can do is upgrade to 2.0 and give it an honest test run -- feedback from actual usage. Of course this is contingent on me releasing the damned docs already! You'll be the first to know when I publish them.

Thank you and happy holidays! :christmas_tree:

**Comment by karfau** [comment]
> The best thing you can do is upgrade to 2.0 and give it an honest test run -- feedback from actual usage. Of course this is contingent on me releasing the damned docs already! You'll be the first to know when I publish them.

That would be great, as long as my current project is still small.
but to stay productive if some things don't work, i think I would like to be able to run both  version in parallel.
(hoping the pages and templating-stuff can stay "mostly the same"...)

Maybe an important part of those docs should be a migration guide, especially running 1.1 as gem and 2.0\* from ruby-code that is checked out from github.
Some hints how this could be accomplished would be great for a ruby-greenhorn like me.
(Maybe this should be a new ticket?)

I'm very excited about whats coming :)

and happy holidays for you to

**Comment by jacobmgreer** [comment]
I also look forward to testing it out once the documentation is ready! :)

**Comment by plusjade** [comment]
@karfau @jacobmgreer you can try running ruhoh 2.0.alpha here: https://github.com/ruhoh/blog/tree/2.0.alpha#readme

Documentation will be up at http://ruhoh.com/edge.

Please test it and let me know if there is anything I missed. Thanks very much!

**Comment by plusjade** [comment]
Closes this in favor of sticking to the previously existing conversation on "full contexts in widgets environment" at #52
